usim/_primitives/context.py

```diff
@@ -228,9 +326,12 @@ def __init__(self, notification: Notification):
         self._notification.__subscribe__(self._activity, self._interrupt)
         return self
 
-    def _handle_exception(self, exc_val) -> bool:
+    def _disable_interrupts(self):
         self._notification.__unsubscribe__(self._activity, self._interrupt)
-        return exc_val is self._interrupt
+        super()._disable_interrupts()
+
+    def _suppress_exception(self, exc_val) -> bool:
```

Comment:
The naming of the method makes me think that it involves actions to *suppress exceptions* and not giving information if exceptions should be suppressed. Please consider renaming.

```suggestion
    def _is_exception_suppressed(self, exc_val) -> bool:
```

Reply:
How about ``_propagate_exception``, ``_is_propagated``, or ``_must_propagate``?

Reply:
I would rather prefer `_is_propagated` as this tells the reader that a `bool` gets returned.